### PR TITLE
feat: support OR logic for visibleForFeatureToggles

### PR DESCRIPTION
**Pull request by DanielSemerdzhievSAP**
Previously, nodes with multiple visibleForFeatureToggles entries required all toggles to be active to be visible (AND logic). This
  change updates the behavior to OR logic — a node is visible if any of the listed feature toggles is active.

  This is needed in hai-mcp-registry-ui where nodes should be visible when either ai-top-level or ai-top-level-registry is applied,
  without requiring both to be present simultaneously.

  Changes:
  - node-utils.service.ts: changed .every() to .some() in isVisible()
  - node-utils.service.spec.ts: updated existing tests to reflect OR semantics and added a new test covering the case where only one
  of multiple toggles is active

## Summary by CodeRabbit

* **Bug Fixes**
  * Navigation nodes now appear when at least one configured feature toggle is active. Nodes with no toggles or an empty toggle list are treated as visible.
* **Tests**
  * Expanded unit tests to cover additional feature-toggle edge cases, including empty toggle lists, negated toggles, and mixed active/inactive scenarios.

### CODE REVIEW

**Review by gkrajniak — CHANGES_REQUESTED** [review]

File: projects/lib/src/lib/services/luigi-nodes/node-utils.service.spec.ts [hunk 1]
```diff
@@ -80,19 +95,33 @@ describe('NodeUtilsService', () => {
...
+        context: {} as NodeContext,
+      };
+      mockLuigiCoreService.isFeatureToggleActive.mockImplementation(
+        (ft) => ft === 'feature1',
+      );
+
+      expect(service.isVisible(node)).toBe(true);
+
+      mockLuigiCoreService.isFeatureToggleActive.mockImplementation(
+        (ft) => ft === 'feature1',
```
[the remark this answers is not in the document]
Comment (DanielSemerdzhievSAP) [reply projects/lib/src/lib/services/luigi-nodes/node-utils.service.spec.ts]: I am not 100% sure what's the best way to test that here. Please feel free to commit, I don't mind.

[the remark this answers is not in the document]
Comment (DanielSemerdzhievSAP) [reply projects/lib/src/lib/services/luigi-nodes/node-utils.service.spec.ts]: Updated, thanks, please take a look

**Review by gkrajniak — APPROVED** [review]
